### DISCUSSION TITLE
Optional affiliate link support?

### DISCUSSION THREAD

**Discussion by poVoq** [bookwyrm-social/bookwyrm#1357] [Ideas]
**Is your feature request related to a problem? Please describe.**
This is probably a controversial suggestion, but what about allowing the addition of affiliate links to places like Amazon?

**Describe the solution you'd like**
This could either be done on user level, thus financially encouraging people to write longer reviews, or (which I personally would find more interesting) on a instance level to help fund running costs.

**Describe alternatives you've considered**
Other easy ways for people to donate would be nice too. Like Patreon links etc.

**Additional context**
I guess it would make sense to first come up with a list of online book vendors that offer this kind of affiliate links?

**Comment by mouse-reeve** [comment]
This is an interesting one! It had crossed my mind as a possibility to add instance-wide affiliate links, but it hadn't occurred to me at the user level. 

At the instance level, I don't see a problem with it (although I'd be really happy to hear objections), although I think it would be worth putting thought into what the best way to disclose to users.

At the user level, it feels dicier to me. My gut reaction is that it would encourage people to try to become an "influencer," and i don't think the behaviors that make someone successful as an influencer necessarily align with the behaviors that make a caring and pleasant community. 

(If it makes a difference to the conversation, there is an [affiliate program through IndieBound](https://www.indiebound.org/affiliate) which appears to be accepting affiliates.)

**Comment by poVoq** [comment]
Hmm, I would have though the user level to be less controversial, given that it wouldn't inject affiliate links into user-posts like an instance wide one would (and that really needs to be very openly communicated).

I get your thought about "influencers", but that seems rather far-fetched to happen in reality on a niche platform like this, no?

What about allowing people to easily add wish-lists to their account, so that others can buy books for them and maybe even get a an cool review in return?

**Comment by mouse-reeve** [comment]
I guess my concept of how instance-level affiliate links would work is that they would exist on the book's landing page, rather than in users' posts. Like having a link in the sidebar to purchase the book, or example. Including them there would definitely be a different proposition.

In terms of influencers, I wonder if even if the audience someone can reach is unlikely to get very large, it still could create a different set of incentives about how to write and interact if there's a monetary reward for getting people to click a link. And since bookwyrm posts can be boosted to mastodon, there can be a larger audience than the scale of bookwyrm itself would suggest -- but certainly it's not even close to the same order of magnitude as, say, a monetized instagram or youtube post. 

I'm vaguely aware that goodreads offers some kind of incentives to reviewers (maybe just free copies of books?), and I'm curious if you (or anyone else) have encountered and had any reaction to that.

**Comment by poVoq** [comment]
No, but I was thinking that authors might be inclined to provide free review copies... at least those self-publishing on sites like Amazon.

**Comment by tastytea** [comment]
Why do people have to earn money through BookWyrm? There are more than enough places for advertisements already.

Additionally, I see these problems with affiliate hyperlinks:

- User-level affiliate hyperlinks incentivize positive reviews, because more people buy good-rated books. I can't trust a review with an affiliate hyperlink.
- I would be angry if something would change my hyperlinks.
- I really don't think we should steer people to user-hostile corporations like Amazon.
- Instance-level affiliate hyperlinks only work with a homogeneous user base. Most book stores focus on one language. Example: I can't find “The Neverending Story” in german on IndieBound.

**Comment by mouse-reeve** [comment]
To be clear, I wouldn't consider an option where users' posts were changed by the system to include affiliate links. And I'm inclined to agree with @tastytea about the incentives that user-level affiliate links would create; I'd rather exist in a community that people are in for the sake of it, without outside financial motives.

I do think it's pretty plausible that an instance would have a homogenous user-base, especially language-wise.
